tests/dashbio_demos/app_alignment_viewer.py

```diff
@@ -587,7 +587,8 @@ def update_storage(dropdown, contents, filename):
         [Input("alignment-chart", "eventDatum")]
     )
     def event_data_select(data):
-        data = json.loads(data)
+        if data is not None:
+            data = json.loads(data)
 
         if data is None or len(data.keys()) == 0:
```

Comment:
```suggestion
        if len(data.keys()) == 0:
```

Reply:
Won't this lead to an error if `data` is `None`? It might be better to move it into the `if` statement above (`if data is not None`) .

Reply:
The case where `data is None` would be handled line 592 (the two pieces of suggestion are meant to go together, they shouldn't be considered/committed independently).

Reply:
Nonetheless, `data` might still not have keys... which *would* lead to an error. We may thus want to refactor this whole check to ensure `data` is a non-empty dictionary.